src/lib/circular-progress/circular-progress-adapter.ts

```diff
@@ -1,56 +1,52 @@
-import { getShadowElement } from '@tylertech/forge-core';
-import { MDCCircularProgress } from '@material/circular-progress';
+import { getShadowElement, toggleAttribute } from '@tylertech/forge-core';
 
 import { ICircularProgressComponent } from './circular-progress';
 import { CIRCULAR_PROGRESS_CONSTANTS } from './circular-progress-constants';
+import { BaseAdapter, IBaseAdapter } from '../core/base/base-adapter';
 
-export interface ICircularProgressAdapter {
-  initialize(): void;
-  destroy(): void;
-  open(): void;
-  close(): void;
+import indeterminateTemplate from './_indeterminate.html';
+import determinateTemplate from './_determinate.html';
+
+export interface ICircularProgressAdapter extends IBaseAdapter {
   setDeterminate(value: boolean): void;
   setProgress(value: number): void;
   setAriaLabel(value: string): void;
 }
 
-export class CircularProgressAdapter implements ICircularProgressAdapter {
+export class CircularProgressAdapter extends BaseAdapter<ICircularProgressComponent> implements ICircularProgressAdapter {
   private _rootElement: HTMLElement;
-  private _mdcCircularProgress: MDCCircularProgress;
+  private _determinateProgressCircleElement: HTMLElement | undefined;
 
-  constructor(private _component: ICircularProgressComponent) {
+  constructor(component: ICircularProgressComponent) {
+    super(component);
     this._rootElement = getShadowElement(this._component, CIRCULAR_PROGRESS_CONSTANTS.selectors.ROOT);
   }
 
-  public initialize(): void {
-    this._mdcCircularProgress = new MDCCircularProgress(this._rootElement);
-  }
-
-  public destroy(): void {
-    this._mdcCircularProgress?.destroy();
-  }
-
   public setDeterminate(value: boolean): void {
-    if (this._mdcCircularProgress) {
-      this._mdcCircularProgress.determinate = value;
+    this._tryResetTemplate();
+    this._rootElement.classList.toggle(CIRCULAR_PROGRESS_CONSTANTS.classes.INDETERMINATE, !value);
+    
+    if (value) {
+      this._rootElement.insertAdjacentHTML('beforeend', determinateTemplate);
```

Comment:
Should we also remove the opposite template if already inserted, e.g. if `determinate` were toggled while already running.  Probably not the most typical usage pattern, but if I'm reading this right, seems you could end up with multiple spinners.

Reply:
I _think_ this is working as expected currently because it calls `this._tryResetTemplate()` which will remove either of the existing templates (indeterminate or determinate). I chose to keep it simple that way because the likelihood of someone toggling between determinate and indeterminate dynamically is probably going to be pretty rare so performance wasn't a concern of mine...

Definitely open to any suggestions you may have though!